lib/ex_unit/lib/ex_unit/runner.ex

```diff
@@ -2,14 +2,18 @@ defmodule ExUnit.Runner do
   @moduledoc false
 
   alias ExUnit.EventManager, as: EM
+  alias ExUnit.RunnerStats
 
   @rand_algorithm :exs1024
 
+  @type manager :: ExUnit.EventManager.manager()
+
   def run(opts, load_us) do
     {:ok, manager} = EM.start_link()
-    {:ok, stats} = EM.add_handler(manager, ExUnit.RunnerStats, opts)
+    {:ok, stat_sup} = EM.add_handler(manager, RunnerStats, opts)
     {opts, config} = configure(manager, opts)
-
+    config = Map.put(config, :stat_sup, stat_sup)
+    config = Map.put(config, :top_pid, self())
```

Comment:
this is the only addition to adds "a feature", we store the top_pid and stat_sup pids, and we avoid to add an extra argument to functions. Maybe there's a better name for top_pid?

Reply:
So lets make sure that we add those when we create the map and not afterwards. Otherwise we get the impression the map is dynamic but it keys are all static (i.e. known upfront). Let's also call those `stats_pid` and `runner_pid`. After all, we don't know that the pid for stats is a sup, that would be leaking implementation detail.

Reply:
done